### ISSUE TITLE
Feature: Hide Sensitive Fields From EEPROM gui

### ISSUE THREAD

**Issue by RadWolfie** [Cxbx-Reloaded/Cxbx-Reloaded#2139]
## Quick summary
Proposal change options for EEPROM's gui window.
1. Add an option for all sensitive fields to reveal from ONE checkbox.
2. Add checkbox next to each sensitive field. (I think all commercial products do use option 2.)
### **or**
3. Reveal field value when it is focused then hide again once unfocused. (easier to manage without need of extra toggle)

## Details
Since we have sensitive fields in EEPROM gui window, by default we don't censor them. Once user report an issue with relative gui window(s), sometimes those screenshots doesn't have censor fields. Either they do not know how to censor them out or don't know if they are meant to be viewed only from their side (without leaking).

See quick summary of proposal change options.

The following fields are critical functional to networking service through LAN or third-party online service.
- HDD Key
- Serial Number
- Online Key
- MAC address

Meanwhile Parental Control Password is not intended to be reveal on the Internet since it fall under personal information privacy.

### System Configuration:
N/A

### Additional Information (if any):
N/A

**Comment by GXTX** [comment]
What do you consider sensitive?

**Comment by RadWolfie** [comment]
- HDD Key
- Serial Number
- Online Key

As for MAC address, I'm not sure if that's sensitive or not. Parental password isn't required unless someone else says otherwise.

**Comment by GXTX** [comment]
These are all auto generated, I'm not sure I see the need to obstruct them behind an additional button.

**Comment by RadWolfie** [comment]
It doesn't matter if they're auto generated, from actual hardware, or manual input, they are ***still*** sensitive fields. Even mac address can't be share with another machine.

Anyway, I think better way to do this without additional button is to have focused field to automatic reveal sensitive info. Once leave the field, revert back to hidden text.

**Comment by GXTX** [comment]
This information isn't tied to users hardware at all, I don't see the point.

**Comment by GXTX** [comment]
Now that there's an open PR for this issue I'll add another comment here.

This information is not used anywhere at all, it's just required for certain kernel functions to be happy. Even if users use their own EEPROM dumps it makes no difference - the information is still not used.

**Comment by x1nixmzeng** [comment]
Can you explain why these are sensitive fields? Like GXTX says, they are automatically generated to keep the kernel happy.

The issue I think you raised is accidentally sharing a screenshot showing fields you have copied from your own hardware, but I'm not sure these should be treated as confidential.

The original Xbox dashboard shows these values in plainsight - what stops somebody sharing a screenshot of that information if they're emulating the dashboard?

**Comment by RadWolfie** [comment]
> Can you explain why these are sensitive fields? Like GXTX says, they are automatically generated to keep the kernel happy.

It's not about make kernel happy, it's about making **titles** compiled with XDK tool happy. Along with homebrew, or official, Xbox live service to work properly.

> The issue I think you raised is accidentally sharing a screenshot showing fields you have copied from your own hardware, but I'm not sure these should be treated as confidential.
> The original Xbox dashboard shows these values in plainsight - what stops somebody sharing a screenshot of that information if they're emulating the dashboard?

They ***are*** confidential from the Internet's eyes. It is require to remain for your eyes only locally or contact support from xbox service requested this info. Even if xbox service are obsolete.

You don't see HDD and Online keys values from Xbox dashboard, they are **secret** keys.

### TLDR Summary:
All fields I mentioned are "critical" fields except for parental control password for different reason. How? They are used for Xbox live service and system link. There are existing third-party tools available to give both xbox hardware and existing emulators communicate over the network. So, they are **not** intended to be exposed onto the Internet. Therefore fall under security breach catalog. Serial Number is used with Xbox live service.

Since users continue to post their EEPROM gui layout on our discord server from time to time. It is decided by the whole team to cover all sensitive fields. In fact, they are not even required to be shown. Other fields are likely need to be show in order to figure out what went wrong. Such as display output or fps issues.

As for the parental control password, simply because... "parental password". Also, why would you want to show it on the Internet? It's for your eyes only as well.

**Comment by x1nixmzeng** [comment]
For my own understanding then, why does the internet take priority over the user?

From a UX point of view, it would make more sense to have a label saying "this information is sensitive, do not share this with others" rather try to treat these fields as passwords, as it isn't super obvious why (at least to a few of us!)
